src/urlencoded.rs

```diff
@@ -1,11 +1,137 @@
-use std::collections::HashMap;
-use url::form_urlencoded;
 use groupable::Groupable;
+use hyper::uri::RequestUri;
+use hyper::uri::RequestUri::{Star, AbsoluteUri, AbsolutePath, Authority};
+use plugin::{Plugin, Pluggable};
+use request::Request;
+use status::StatusCode;
+use std::collections::HashMap;
+use std::io::{Error, Read};
+use typemap::Key;
+use url::{form_urlencoded, UrlParser};
+
+type QueryStore = HashMap<String, Vec<String>>;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct Query(QueryStore);
```

Comment:
My concern is for users reading the return type in their rustdoc output, it may be confusing if they see `form_data(..) -> &Query` rather than `form_data(..) -> &FormParams`. Or in the case where the compiler is throwing type errors with `Query` in it when they think they are not dealing with queries.